### PR TITLE
Autogen keys

### LINKED ISSUE #203: Allow public keys to be specified at build time
Instead of baking the specific public keys directly into the build, allow cmake variables (for Zephyr) to point to a file containing the keys to be built in.

### PR DESCRIPTION
**Pull request by utzig**
Add support for describing the key file in the configuration.

Feature complete but might be in need of some cleanups. Suggestions are welcomed!

This fixes #203

### CODE REVIEW

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -36,6 +36,23 @@ config BOOT_SIGNATURE_TYPE_ECDSA_P256
 
 endchoice
 
+config HAVE_SIGNATURE_KEY_FILE
```
[the remark this answers is not in the document]
Comment (utzig) [reply boot/zephyr/Kconfig]: I suppose that's what I wanted... which is why this still need "style" fixes ;-)

File: boot/zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -120,3 +120,23 @@ zephyr_sources(${BOOT_DIR}/zephyr/serial_adapter.c)
 
 add_subdirectory(${BOOT_DIR}/boot_serial ./boot/boot_serial)
 endif()
+
+if(CONFIG_HAVE_SIGNATURE_KEY_FILE)
+  set(KEY_FILE ${MCUBOOT_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
+  if(NOT EXISTS ${KEY_FILE})
+    message(FATAL_ERROR "Could not find key file: '${KEY_FILE}'.")
+  endif()
+  set(GENERATED autogen-pubkey.c)
```
[the remark this answers is not in the document]
Comment (utzig) [reply boot/zephyr/CMakeLists.txt]: yes!

**Comment by utzig** [comment]
@mbolivar @d3zd3z @carlescufi I have been using this for the last 2 weeks, is something still missing here, any style changes I have to make?

**Comment by mbolivar** [comment]
> @mbolivar @d3zd3z @carlescufi I have been using this for the last 2 weeks, is something still missing here, any style changes I have to make?

Hmm, looks like there is still cleanup work to do:

cb3164e Fix some review issues - rebase/reword later

Did you mean to do that to get re-review?

FWIW, it makes it take longer (for me at least, and I think this is common) to review PRs if I have to look at the entire series as a whole because there are "fix it" commits added on to the end. It's much easier (and thus faster) to read and review a series when the patches form a logical progression of isolated changes.

This style of doing the rebasing and cleanup as part of the updated pull request is the Linux kernel and Zephyr style of sending in patches, at least, and I guess it's common for other open source projects as well. GitHub kind of discourages this style in some ways with their pull request interface, but IMHO it's a mistake: optimizing pull requests to use less review bandwidth is more efficient for the project, since that's always what's limited.

**Comment by utzig** [comment]
@mbolivar Fixed.

**Comment by utzig** [comment]
@mbolivar @d3zd3z @carlescufi Any other feedback here? Should I just merge?

**Pull request merged**